docs/API.md

```diff
@@ -0,0 +1,176 @@
+* 404 Not found  
+If a variable in the path doesn't reference a valid resource 
+* 503 Service Unavailable  
+Error if an upstream system cannot be reached (e.g. Deckhand for documents, Airflow for actions)
+
+---
+## <a name="DocumentStagingAPI"></a> Document Staging API
+Shipyard will serve as the entrypoint for documents and secrets into a site. At any point in time, there will be two represented versions of documents in a site that are accessible via this API. 
+
+* The lastDeployed version, which represents the last version of documents that successfully completed deployment.
+* The staged version, which represents the document set that has been ingested by this API since the lastDeployed version.
+
+### /v1.0/configDocs  
+Represents the site configuration documents.  
+
+#### POST 
+Ingests site configuration documents. Synchronous.
+##### Responses
+* 200 OK  
+If the documents are successfully ingested, even with validation failures. Response message includes:
+  * a link to the GET /configDocs 
+  * a list of validation results
+
+#### GET 
+Returns the source documents for the most recently staged version
+Query Parameter "lastDeployed=true" will insetead return the documents for the currently deployed version.
+##### Responses
+* 200 OK  
+If documents can be retrieved.
+
+#### DELETE  
+Updates the configDocs to be the lastDeployed versions in deckhand, effectively discarding any staged documents.
+###### Responses
+* 204 No Content
+
+
+### /v1.0/secrets
+Represents the secrets documents that will be used in combination with the config docs.  
+
+##### POST 
+Ingest new secrets documents.
+###### Responses
+201 Created  
+If the documents are successfully ingested, even with validation failures. Response message includes:
+  * a link to the GET /configDocs
+  * a list of validation results
+
+#### GET 
+Return the secrets documents
+##### Responses
+* 200 OK  
+If documents can be retrieved.
+
+#### DELETE
+Updates the secrets to be the lastDeployed versions in deckhand, effectively discarding any staged documents.
+##### Responses
+* 204 No Content
+
+---
+## <a name="ActionAPI"></a> Action API
+Signal to Shipyard to take an action (e.g. Invoke a named DAG)
+See [Action Commands](API_action_commands.md) for supported actions
+
+
+### /v1.0/actions
+All actions will include fields that indicate the following data:
+* id (assigned during POST)
+* name - the name of the action - likely to be the DAG Names, but may be mapped for "nicer" values.
+* parameters - a dictionary of the parameters configuring the action.
+* tracking info - (user, time, etc)
+* action_lifecycle value:
+  * Pending
+  * Validation Failed
+  * In DAG
+  * Complete
+  * Failed
+* DAG_status - representing the status that airflow provides for an executing DAG
+* validations - a list of validaitons that have been done, including any status information for those validations. During the lifecycle of the DAG, this list of validations may continue to grow.
```

Comment:
Is it possible to pre-populate this validation list so that as soon as the action starts there is a reference list of all the validations? Initially they can start as 'pending' or something.